creme/neighbors/base_neighbors.py

```diff
@@ -0,0 +1,187 @@
+import itertools
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+from creme.utils.skmultiflow_utils import get_dimensions
+
+
+class KNeighborsBuffer:
+    """Keep a fixed-size sliding window of the most recent data samples.
+
+    Parameters
+    ----------
+    window_size : int, optional (default=1000)
+        The window's size.
+
+    Raises
+    ------
+    ValueError
+        If at any moment, a sample with a different number of attributes than
+         those already observed is passed.
+
+    Notes
+    -----
+    It updates its stored samples by the FIFO method, which means
+    that when size limit is reached, old samples are dumped to give
+    place to new samples.
+
+    The internal buffer does not keep order of the stored samples,
+    when the size limit is reached, the older samples are overwritten
+    with new ones (circular buffer).
+
+    """
+
+    def __init__(self, window_size=1000):
+        self.window_size = window_size
+        self._n_features = -1
+        self._n_targets = -1
+        self._size = 0
+        self._next_insert = 0
+        self._oldest = 0
+        self._imask = None
+        self._X = None
+        self._y = None
+        self._is_initialized = False
+
+    def _configure(self):
+        # Binary instance mask to filter data in the buffer
+        self._imask = np.zeros(self.window_size, dtype=bool)
+        self._X = np.zeros((self.window_size, self._n_features))
+        self._y = [None for _ in range(self.window_size)]
+        self._is_initialized = True
+
+    def reset(self):
+        """Reset the sliding window. """
+        self._n_features = -1
+        self._n_targets = -1
+        self._size = 0
+        self._next_insert = 0
+        self._oldest = 0
+        self._imask = None
+        self._X = None
+        self._y = None
+        self._is_initialized = False
+
+        return self
+
+    def add_one(self, x, y):
```

Comment:
I would call this `append` to follow the `collections.deque` terminology.

Reply:
Nice catch! I changed it in [1a91c21](https://github.com/creme-ml/creme/pull/323/commits/1a91c2109076c5b00b947242115ed4f346681c42). Maybe the current usage is confusing, since we use `append` and pass separate `x` and `y` objects.